staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -770,6 +850,15 @@ func (o *DiffOptions) Run() error {
 
 // Validate makes sure provided values for DiffOptions are valid
 func (o *DiffOptions) Validate() error {
+	if o.applySet != nil {
+		// if !o.Prune {
```

Comment:
why commented out?

Reply:
So I wasn't sure whether `kubectl diff` should require the `--prune` flag.  We can show a diff without doing the prune preview.  But I'm also not sure whether we should default to prune=true or prune=false if it is not specified, so I'm going to go with our "make it restrictive at first and relax it where requested" approach; so I'm adding this back in.